### ISSUE TITLE
`use $crate::{self};` should not be accepted

### ISSUE THREAD

**Issue by theemathas** [rust-lang/rust#146967]
tried this code:

```rust
macro_rules! foo {
    () => {
        use $crate::{self};
    };
}

foo!();
```

I expected a compile error of some sort. Instead, it compiles fine.

Based on the error message when I try to import it twice, I'm guessing that it's importing the current `crate` with the name `$crate`.

<details>
<summary>Importing it twice</summary>

```rust
macro_rules! foo {
    () => {
        use $crate::{self};
    };
}

foo!();
foo!();

fn main() {}
```

```
error[E0252]: the name `$crate` is defined multiple times
 --> src/main.rs:3:22
  |
3 |         use $crate::{self};
  |         -------------^^^^--
  |         |            |
  |         |            `$crate` reimported here
  |         |            previous import of the module `$crate` here
  |         help: remove unnecessary import
...
7 | foo!();
  | ------ in this macro invocation
  |
  = note: `$crate` must be defined only once in the type namespace of this module
  = note: this error originates in the macro `foo` (in Nightly builds, run with -Z macro-backtrace for more info)
```

</details>

I am also able to use this import via the name `$crate`:

<details>
<summary>Using the import</summary>

```rust
macro_rules! foo {
    () => {
        use $crate::{self};
        use crate::$crate as lol;
    };
}

foo!();

fn wtf() {
    println!("wtf");
}

fn main() {
    lol::wtf();  // this compiles
}
```

</details>

Surely, this `$crate` identifier should not be able to be used as an actual name?

### Meta

Reproducible on the playground with version `1.92.0-nightly (2025-09-23 975e6c8fec280816d24f)`

**Comment by petrochenkov** [comment]
Duplicate of one of the issues listed in https://github.com/rust-lang/rust/issues/119776#issuecomment-1883406204.
Imports should always end up having a name (https://github.com/rust-lang/rust/issues/35612#issuecomment-578425321).
